### ISSUE TITLE
ClimateMachine init keyword arg defaults

### ISSUE THREAD

**Issue by akshaysridhar** [CliMA/ClimateMachine.jl#1317]
# Description

Is there a reason for `disabling` command line arguments by default ? 

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

**Comment by jakebolewski** [comment]
When using climate machine programmatically you don't want to greedily parse all command line ARGS of a process as then the initialization doesn't compose well (at all).

The CLI behavior is fine for top level scripts, but it's not great for running things in a REPL / notebook / tests / recursive invocation, etc.

I would think that the main use case would be interactive (in a REPL / tutorial / notebook), but this might not be correct.

@simonbyrne, @charleskawczynski, @kpamnany  might have other opinions.

**Comment by jakebolewski** [comment]
Basically under the assumption that side effects should be explicit and not implicit behavior.

**Comment by smarras79** [comment]
For practical purposes, I think that if the user sets `--diagnostics INTERVAL` and/or `--vtk INTERVAL`, the code should always produce an output regardless of how the code is executed. Even if output affects performance, there is really no way that we should provide a code that requires knowledge of the code to make sure that the output is written  to file.

**Comment by jakebolewski** [comment]
@smarras79 I think that is a separate issue.

**Comment by smarras79** [comment]
> @smarras79 I think that is a separate issue.

I agree with that, @jakebolewski , but the issue was brought up by @akshaysridhar because this change prevented the output from being written to file.

**Comment by kpamnany** [comment]
The question is really about the _default_ behavior, i.e. whether experiments need to use `ClimateMachine.init(parse_clargs = true)` or whether tests/notebooks/etc. need to use `ClimateMachine.init(parse_clargs = false)`.

I don't think the PR 1303 change really affects this because the question just becomes: 'which ClimateMachine initialization function should I be using?' But I don't mind reverting that change if it will make things easier.

**Comment by kpamnany** [comment]
Actually there's a second question, which applies only to diagnostics at the moment. If a driver configures a diagnostics group and passes in a diagnostics configuration, it is ignored _unless_ the `diagnostics` command line argument or initialization keyword argument is provided. This too, is a default that can be changed if desired.
